src/route/page.rs

```diff
@@ -18,37 +18,134 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
-use super::prelude::*;
+use crate::{remote::DeepwellPool, route::prelude::*};
+use actix_web::{Error, *};
+use deepwell_core::prelude::*;
+use ftml::{
+    data::User as FtmlUser,
+    handle::{NullHandle, RemoteHandle},
+    prefilter,
+    prelude::*,
+    RemoteError, RemoteResult,
+};
 use std::collections::HashMap;
 use wikidot_path::Request as PageRequest;
 
 // Public route methods
 
+/* TODO: Temporary Measure
+ *
+ * Until we can figure out a good way of divining the WikiId from the wiki request,
+ * just assume that these functions are retrieving the wiki at WikiId 1.
+ */
+lazy_static! {
+    static ref TEMP_WIKI_ID: WikiId = WikiId::from_raw(1);
+}
+
+/// Return an HttpResult for getting a page by its slug, using deepwell.
+pub async fn get_deepwell_page(
+    wiki_id: WikiId,
+    slug: &str,
+    deepwell: &web::Data<DeepwellPool>,
+) -> std::result::Result<HttpResponse, ()> {
+    debug!("Retrieving page by WikiId {} and slug {}", wiki_id, slug);
+    let result = deepwell
+        .claim()
+        .await
+        .get_page_contents(wiki_id.clone(), slug.to_string())
+        .await;
+
+    match try_io_result!(result) {
+        Ok(Some(page)) => {
+            // now run FTML on it
+            let mut contents = match String::from_utf8(page.to_vec()) {
```

Comment:
Hmm, we'll need to check if it's safe to switch from `Box<[u8]>` to `String`. I'll check if wikidot has any non-UTF-8 pages.